### DISCUSSION TITLE
Deprecation Notice

### DISCUSSION THREAD

**Discussion by mdiddie** [community/community#23933] [New to GitHub]
<p>Hello - I’m fairly new to HTML/CSS/Javascript. I’m using Visual Studio Code to write and then publish my content. I’m not really sure what the deprecation notice means or what I need to do so that I can still upload my code through VSC. I’d love any help or being pointed to helpful resources for understanding this change. Thanks in advance!</p>

**Comment by airtower-luna** [comment]
<p>Could you quote the deprecation notice, so we know what it’s about? From the context I guess the password deprecation for Git and API access?</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/mdiddie">@mdiddie</a> you could take a look at this post and links to see it is similar, guessing you are talking about password depracation notices also and to switch to personal access tokens or SSH key for authentication</p><aside class="quote quote-modified" data-post="4" data-topic="142959">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/basic-authentication-using-a-password-to-the-api/142959/4">Basic authentication using a password to the API</a> <a class="badge-wrapper  bullet" href="/c/github-ecosystem/37"><span class="badge-category-bg" style="background-color: #28a745;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and support using GitHub’s REST, and GraphQL APIs, building Applications and OAuth Apps, the GitHub Marketplace, and third party integrations.">GitHub Ecosystem</span></a>
  </div>
  <blockquote>
    <a class="mention" href="/yannberry">@YannBerry</a>, <a class="mention" href="/saya7852">@saya7852</a> 
The link to 2020-02-14-deprecating-password-auth is a valid GitHub change notification. 
The email looks valid (and similar to other users I have seen asking about, it is not asking you to click on any suspicious links. 
You should switch to using personal access token, further details here 

You mention “As far as I’m concerned I don’t use the API. Why do I receive this email ? What should I do ?” 
If that is the case I would change my password immediately (and enable 2FA…
  </blockquote>
</aside>

**Comment by mdiddie** [comment]
<p>Hi! So I received an email and it contained this link:</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener nofollow ugc" title="08:28PM - 15 December 2020">The GitHub Blog – 15 Dec 20</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/09/security-1200-630.png?fit=1200%2C630" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener nofollow ugc">Token authentication requirements for Git operations - The GitHub Blog</a></h3>

<p>Beginning August 13th, 2021, we will no longer accept account passwords when authenticating Git operations on GitHub.com and will instead require token-based authentication (for example, a personal access, OAuth, or GitHub App installation token) for...</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I’m pretty overwhelmed by the information provided - I had a friend help me connect my Visual Studio Code account to my Github account. I’m not a programmer by any means! I guess I’m just trying to figure out what this means in regards to how I upload my content through VSC? Do I need to change something?</p>

**Comment by mdiddie** [comment]
<p>Okay, so you are suggesting I should switch to a ‘personal access token.’ I have no idea what that is but I’ll look it up - if you have a good resources for learning more about what these tokens are and how to use them I’d be really appreciative! Thankyou!</p>

**Comment by mdiddie** [comment]
<p>So, does this seem right? First, I need to make a PAT:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" target="_blank" rel="noopener nofollow ugc">Creating a personal access token - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>And then I need to apply it to my VSC:</p>
<p><a href="https://marketplace.visualstudio.com/items?itemName=KnisterPeter.vscode-github#:~:text=Setup%20Personal%20Access%20Token&amp;text=To%20execute%20the%20'GitHub%3A%20Set,the%20token%20generated%20from%20GitHub" class="inline-onebox" rel="noopener nofollow ugc">GitHub - Visual Studio Marketplace</a>.</p>

**Comment by airtower-luna** [comment]
<p>Basically you just need to use the PAT, as <a class="mention" href="/byrneh">@byrneh</a>’s pointed out. This means to create a PAT (you already found the documentation on that) and use that instead of your password when VSC connects to Github. Maybe someone who’s familiar with VSC can describe how to do that?</p>

**Comment by CerebralDatabank** [comment]
<p>Create the Personal Access Token by going to Profile Icon on GitHub page &gt; Settings &gt; Developer settings &gt; Personal access tokens and name it “Home PC VS Code” (or whatever you want). Leave that page open.</p>
<p>Go to VS Code and click the user profile symbol in the bottom left (above the settings gear icon).</p>
<p>Select the “GitHub” option, enter your username/email as normal, and copy-and-paste the Personal Access Token you created in the password field.</p>
<p>After making sure it signed in successfully, close the tab you left open with the Personal Access Token (you won’t be able to see it again).</p>
<p>Hope that helped :slight_smile:</p>

**Comment by byrneh** [comment]
<p>I am not an active VSC, but checked the latest versions of Visual Studio Code 1.8.0 (January 26, 2021) and Git version 2.30.0.windows.2 installed.</p>
<p>I did a simple clone of a private repository and it successfully worked with the option to [Sign in with your browser].  when you click on this option it should launch an authentication flow in your default browser (you may need to manually switch if the focus does not swap automatically to the launched browser tab).<br>
in the visual studio terminal window you will see message<br>
info: please complete authentication in your browser…</p>
<p>In the launched browser tab you then simply confirm if already logged in or GitHub login as usual if not currently logged in to GitHub in your default browser.</p>
<p>It will create and cache a token behind the scenes for you, which seems the easiest for an inexperienced or user or someone unfamiliar or not wanting to configure and manage personal access tokens and SSH keys for Git access.</p>
<p>NOTE: If you want us use GitHub’s REST/GRAPHQL APIs you will have to use get used to using personal access tokens.</p>

**Comment by StanleyMasinde** [comment]
<p>It means that typing your password on the terminal will not be supported.<br>
You can either use a <a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" rel="noopener nofollow ugc">personal access token</a><br>
or <a href="https://github.com/settings/ssh/new" rel="noopener nofollow ugc">SSH keys</a> (I prefer this)</p>

**Comment by mdiddie** [comment]
<p>What is preferable about the SSH keys?</p>

**Reply by StanleyMasinde** [reply]
I must've missed this. What did you use these days?

**Comment by byrneh** [comment]
<p><a class="mention" href="/mdiddie">@mdiddie</a>, its a matter of personal choice in many ways. Its not uncommon for people who do not understand SSH and configuration to experience problems on setup or at some point later in time.<br>
SSH can only be used for Git Actions, a personal access token is required for API access.</p>
<p>For SSH keys used by a human(user), as opposed to configured in server (machine account), you have the recommended option of adding an SSH passphrase, which adds an additional level of security should a malicious actor obtain your SSH private key.</p>
<p>There is a lot more that could be said, discussed and argued on the respective merits of the various authentication methods for each usage scenario. It would be large topic in itself to discuss.</p>
<p>The web auth flow (sign in with browser) seems the simplest for the beginner to be using.</p>
<p>Are you now working ok or still stuck after the posted replies above?</p>

**Comment by StanleyMasinde** [comment]
<p>I love it because I don’t have to input my password every-time i push or pull the same applies to my automated deployments on my server.</p>
<p>as <a class="mention" href="/byrneh">@byrneh</a> has mentioned you can use a SSH Key with a passphrase but I personally don’t like that.</p>

**Comment by mdiddie** [comment]
<p>I haven’t done anything yet because I haven’t had time to go through the steps! But thanks for checking in. :slight_smile:</p>

**Comment by TLBauer** [comment]
<p>I’ve got my PAT.  How do I go about plugging it in to Eclipse.  Surely I do not have to reset every password on every repository?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/c/bc14eb76e8b63e899e58041a7839b83601db1e33.png" data-download-href="/uploads/short-url/qPQAgfDR5e0fTRCpc6DntREmaxt.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/25821777/181080278-0f24333f-a065-4a8a-91a5-8117fabcdf1b.png" alt="image" data-base62-sha1="qPQAgfDR5e0fTRCpc6DntREmaxt" width="690" height="250" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/c/bc14eb76e8b63e899e58041a7839b83601db1e33_2_690x250.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/c/bc14eb76e8b63e899e58041a7839b83601db1e33_2_1035x375.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/c/bc14eb76e8b63e899e58041a7839b83601db1e33.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/c/bc14eb76e8b63e899e58041a7839b83601db1e33_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1167×424 61.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mdiddie** [comment]
<p>Thanks for this! I don’t actually see a ‘user profile’ icon in my Visual Studio Code platform above the gear icon on the bottom left. I’m still trying to look around for the ‘user profile.’ There are ‘User settings’ but I went through those and I still don’t see anything about a ‘Github’ option where I can enter a username/email. Any further help would be appreciated! I’ll keep looking.</p>
<p>Found this:<br>
<a href="https://marketplace.visualstudio.com/items?itemName=KnisterPeter.vscode-github" class="onebox" target="_blank" rel="noopener nofollow ugc">https://marketplace.visualstudio.com/items?itemName=KnisterPeter.vscode-github</a></p>
<p>I tried that resource but have hit another brick wall. I made the PAT in Github but I cannot figure out how to apply it in VSC. I tried typing in “GitHub: Set Personal Access Token” in the command palette in VSC but nothing comes up and I don’t even see a user profile or anywhere I might be using a password. Maybe I don’t need to use the PAT in VSC? Also, after I created the PAT in Github I made a change to my website and pushed the request through to my Github website - it worked fine. Maybe it’s okay?</p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="mdiddie" data-post="17" data-topic="160385">
<div class="title">
<div class="quote-controls"></div>
 mdiddie:</div>
<blockquote>
<p>I tried that resource but have hit another brick wall. I made the PAT in Github but I cannot figure out how to apply it in VSC. I tried typing in “GitHub: Set Personal Access Token” in the command palette in VSC but nothing comes up and I don’t even see a user profile or anywhere I might be using a password. Maybe I don’t need to use the PAT in VSC? Also, after I created the PAT in Github I made a change to my website and pushed the request through to my Github website - it worked fine. Maybe it’s okay?</p>
</blockquote>
</aside>
<p><a class="mention" href="/mdiddie">@mdiddie</a> , VSC uses Git, so I would expect it to behave the same as Git Authentication and caching in Credential Manager.  So prompting you to Signed in if not already signed in.<br>
There is also some <a href="https://code.visualstudio.com/docs/editor/github" rel="noopener nofollow ugc">Visual Studio documentation</a></p>

**Comment by mdiddie** [comment]
<p>I’m actually not having any trouble uploading code to Github through VCS. I have created a PAT on my Github account and I’m trying to figure out if I need to change anything on my VCS account - because of the deprecation notice. So far, VCS doesn’t seem to be affected by the new PAT so I’m thinking maybe it’s good.</p>

**Comment by CerebralDatabank** [comment, marked as answer]
<p>Came across this thread again; it’s been a while, but I’ll leave this here for future readers.</p>
<p>I just installed VS Code in a fresh Windows Sandbox instance, and the user profile icon is present above the settings icon - make sure you have the latest version, there are no potentially conflicting extensions, and the icon is not hidden (Right click on an empty spot in the sidebar and make sure all of the items are checked, especially “Accounts”).</p>
<p><img src="https://user-images.githubusercontent.com/24465111/181080283-ed568d9a-24ef-4700-810d-dab4b2cf5004.png" alt="image" data-base62-sha1="yI85WLk2HJrXiZuF8OpGQqC0GdT" width="73" height="154"><br>
<em>User profile icon</em></p>
<p>However, the user profile icon did not have a GitHub option - to log in, I selected the Source Control icon in the sidebar, clicked Clone Repository, chose the “Clone from GitHub” option, clicked Allow on the dialog asking to sign in, and authorized GitHub through the browser. Then, my GitHub account appeared when clicking the user profile icon. (So, it seems like a PAT is no longer necessary.)</p>
<p>There is another extension called “GitHub Pull Requests and Issues” available separately in the extension store; I’m pretty sure it will just use the account details you already have if you’re signed in to GitHub.</p>

**Comment by mdiddie** [comment]
<p>This was it! I need to update Visual Studio Code and then the profile icon appeared and I was able to sync the settings and connect to Github with the PAT. Thank you so much!</p>
